http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -390,37 +390,13 @@ class ClusterMonitorActor(
           Future.successful(NotReadyCluster(googleStatus, googleInstances))
         // Take care we don't restart a Deleting or Stopping cluster if google hasn't updated their status yet
         case Running
-            if leoClusterStatus != Deleting && leoClusterStatus != Stopping && leoClusterStatus != Starting && runningInstanceCount == googleInstances.size =>
-          getMasterIp.map {
-            case Some(ip) => ReadyCluster(ip, googleInstances)
-            case None     => NotReadyCluster(ClusterStatus.Running, googleInstances)
-          }
-        case Running if leoClusterStatus == Starting && runningInstanceCount == googleInstances.size =>
+            if leoClusterStatus != Deleting && leoClusterStatus != Stopping && runningInstanceCount == googleInstances.size =>
           getMasterIp.flatMap {
-            case Some(ip) =>
```

Comment:
The meat of the fix. Previously we were only checking `ClusterTool` status in the transition from `Starting` -> `Running`. I lifted that code into its own method and now call it for both `Starting` -> `Running` and `Creating` -> `Running` status transitions. I also collapsed some redundant `case` statements.